### PR TITLE
functions and rollout fix

### PR DESCRIPTION
**Pull request by Hashfyre**
- rewrites the entire script using bash-functions
- adds the `pollDeploymentRollout` function to poll status without `watch` every 10 secs

todo:
 - add checks for  failure conditions to the `kubectl rollout status`'s result
   - kubectl rollout status doesn't yet have a machine readable output with either `-o yaml | json`
   - need to verify if I can just bubble the exit code.
 - make the sleep param configurable as a `plugin_variable`(subject to opionion)

### CODE REVIEW

File: update.sh [hunk 1]
```diff
@@ -1,99 +1,219 @@
...
+  done
+}
+
+startDeployment(){
+  local CLUSTER=$1; shift
+  local NAMESPACE=$1; shift
+  local DEPLOYMENTS=$1; shift
+  local CONTAINERS=$1
+
+  for DEPLOY in ${DEPLOYMENTS[@]}; do
```
Comment (gappan) [inline update.sh]: This does not seem to be running. We tried to deploy lumberjack [Ref](https://drone.razorpay.com/razorpay/lumberjack/38) to multiple deployments. but it skipped the 2 nd deployment `lumberjack-sqs`

**Review by gappan — CHANGES_REQUESTED** [review]
Please check the functionality for multiple deployments

File: update.sh (hunk 1 above)
Comment (Hashfyre) [reply update.sh]: Thanks for the report.

**Comment by Hashfyre** [comment]
@gappan The issue with multiple deployment functionality has been fixed.
screen shot 2018-01-30 at 8 40 07 pm

**Pull request merged**